astropy/table/sorted_array.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+
+class SortedArray(object):
+    '''
+    Implements a sorted array container using
+    a list of numpy arrays.
+
+    Parameters
+    ----------
+    lines : `Table`
```

Comment:
> Yeah, splitting it that way is probably best. It's necessary to have the row column bundled with the value columns in a Table before performing lexsort, but after that we can split it up again into data and row_index when passing to the engine __init__.

Is that because you are worried about making the sort be stable?  Otherwise I don't see why you need the row as part of the sort.